src/test/java/codeu/model/store/basic/NotificationStoreTest.java

```diff
@@ -0,0 +1,169 @@
+    assertEquals(NOTIFICATION_TWO, resultNotifications.get(0));
+
+    resultNotifications = notificationStore.getNotificationsForUser(USER_TWO);
+    assertEquals(NOTIFICATION_THREE, resultNotifications.get(0));
+  }
+
+  @Test
+  public void testGetReadNotificationsForUser() {
+    notificationStore.markNotificationAsViewed(NOTIFICATION_ONE);
+    List<Notification> resultNotifications = notificationStore.getReadNotificationsForUser(USER_ONE);
+    Assert.assertEquals(1, resultNotifications.size());
+    assertEquals(NOTIFICATION_ONE, resultNotifications.get(0));
+  }
+
+  @Test
+  public void testGetNumNotificationsForUser() {
+    List<Notification> resultNotifications = notificationStore.getNotificationsForUser(USER_ONE);
+    Assert.assertEquals(2, resultNotifications.size());
+  }
+
+  @Test
+  public void testGetNumUnreadNotificationsForUser() {
+    List<Notification> resultNotifications = notificationStore.getNotificationsForUser(USER_ONE);
+    resultNotifications.get(0).markAsViewed();
+    int numUnreadNotifications = notificationStore.getNumUnreadNotificationsForUser(USER_ONE);
+    Assert.assertEquals(notificationStore.getNumUnreadNotificationsForUser(USER_ONE), numUnreadNotifications);
+  }
+
+  @Test
+  public void testGetNumReadNotificationsForUser() {
+    notificationStore.markNotificationAsViewed(NOTIFICATION_ONE);
+    List<Notification> resultNotifications = notificationStore.getReadNotificationsForUser(USER_ONE);
+    Assert.assertEquals(notificationStore.getNumReadNotificationsForUser(USER_ONE), resultNotifications.size());
+  }
+
+  @Test
+  public void testDeleteNotification() {
+    final List<Notification> notificationList = new ArrayList<>();
+    notificationList.add(NOTIFICATION_ONE);
+    notificationList.add(NOTIFICATION_TWO);
+    notificationStore.setNotifications(notificationList);
+    notificationStore.deleteNotification(NOTIFICATION_ONE);
+
+    List<Notification> resultNotifications = notificationStore.getNotificationsForUser(USER_ONE);
+    Assert.assertEquals(1, resultNotifications.size());
+    assertEquals(NOTIFICATION_TWO, resultNotifications.get(0));
+  }
+
+  @Test
+  public void testDeleteAllNotificationsForUser() {
+    Assert.assertEquals(2, notificationStore.getNotificationsForUser(USER_ONE).size());
+    notificationStore.deleteAllNotificationsForUser(USER_ONE);
+    Assert.assertEquals(0, notificationStore.getNotificationsForUser(USER_ONE).size());
+  }
+
+  private void assertEquals(Notification expectedNotification, Notification actualNotification) {
+    Assert.assertEquals(expectedNotification.getId(), actualNotification.getId());
+    Assert.assertEquals(expectedNotification.getNotifiedUserUUID(), actualNotification.getNotifiedUserUUID());
+    Assert.assertEquals(expectedNotification.getMessageUUID(), actualNotification.getMessageUUID());;
+    Assert.assertEquals(expectedNotification.getCreationTime(), actualNotification.getCreationTime());
+  }
```

Comment:
all of these tests look great!! all of the test names are very clear and so are all your variables